public/js/index.js

```diff
@@ -128,6 +128,17 @@ $(function () {
         console.error("Error saving config:", strError);
       }
     });
+
+  })
+
+  $('#updateSoftware').on('click', function () {
+    if (objSocket.connected) {
+      objSocket.emit('control_hat', '{“set_charger”: “ON”}')
+      objSocket.emit('update', `{"timestamp":"${Date.now()}", "version":"${RQ_PARAMS.UPDATE_VERSION}", "action":"UPDATE", "args":"UI"}`)
```

Comment:
Date.now() returns milliseconds. In another module in the ui code we have timestamp_ms which is milliseconds.
Please either divide Date.now() by 1000 and round() of let's rename to timestamp_ms. I prefer the former.

Reply:
done